packages/clay-css/src/scss/cadmin/components/_treeview.scss

```diff
@@ -0,0 +1,421 @@
+
+				@include clay-css($focus);
+
+				.component-expander {
+					$component-expander: setter(
+						map-get($focus, component-expander)
+					);
+
+					@include clay-css($component-expander);
+				}
+			}
+		}
+
+		.component-expander {
+			$component-expander: setter(
+				map-get(
+					$cadmin-treeview-show-component-expander-on-hover,
+					component-expander
+				),
+				()
+			);
+
+			@include clay-css($component-expander);
+		}
+	}
+}
+
+.treeview-group {
+	$group: setter(map-get($cadmin-treeview, group), ());
+
+	@include clay-css($group);
+}
+
+.treeview-item {
+	$item: setter(map-get($cadmin-treeview, item), ());
+
+	@include clay-css($item);
+
+	&:last-child {
+		$last-child: setter(map-get($item, last-child), ());
+
+		@include clay-css($last-child);
+	}
+}
+
+.treeview-link {
+	$link: setter(map-get($cadmin-treeview, link), ());
+
+	@include clay-link($link);
+
+	&.hover,
+	&:hover,
+	&.focus,
+	&:focus {
+		.component-action {
+			display: block;
+		}
+	}
+
+	&.show,
+	&[aria-expanded='true'] {
+		.component-expander {
+			.component-expanded-d-none {
+				display: none;
+			}
+
+			.lexicon-icon:not(.component-expanded-d-none) {
+				display: inline-block;
+			}
+		}
+	}
+}
+
+.component-expander {
+	$component-expander: setter(
+		map-get($cadmin-treeview, component-expander),
+		()
+	);
+
+	@include clay-button-variant($component-expander);
+
+	.lexicon-icon:not(.component-expanded-d-none) {
+		display: none;
+	}
+}
+
+.component-action {
+	$component-action: setter(map-get($cadmin-treeview, component-action), ());
+
+	@include clay-button-variant($component-action);
+}
+
+.component-icon {
+	$component-icon: setter(map-get($cadmin-treeview, component-icon), ());
+
+	@include clay-css($component-icon);
+
+	.lexicon-icon {
+		$lexicon-icon: setter(map-get($component-icon, lexicon-icon), ());
+
+		@include clay-css($lexicon-icon);
+	}
+}
+
+.component-text {
+	$component-text: setter(map-get($cadmin-treeview, component-text), ());
+
+	@include clay-css($component-text);
+}
+
+.treeview-nested-margins {
+	.treeview-group .treeview-item {
+		margin-left: 24px;
+	}
+}
+
+.treeview-nested {
+	padding-bottom: 4px;
+	padding-top: 4px;
+
+	> .treeview-item > .treeview-link {
+		padding-left: 0;
+
+		> .c-inner {
+			margin-left: 0;
+		}
+	}
+
+	.treeview-group {
+		> .treeview-item > .treeview-link {
+			padding-left: 24px;
+
+			> .c-inner {
+				margin-left: -24px;
+			}
+		}
+	}
+
+	.treeview-group .treeview-group > .treeview-item > .treeview-link {
+		padding-left: 48px;
+
+		> .c-inner {
+			margin-left: -48px;
+		}
+	}
+
+	.treeview-group
+		.treeview-group
+		.treeview-group
+		> .treeview-item
+		> .treeview-link {
+		padding-left: 72px;
+
+		> .c-inner {
+			margin-left: -72px;
+		}
+	}
+
+	.treeview-group
+		.treeview-group
+		.treeview-group
+		.treeview-group
+		> .treeview-item
+		> .treeview-link {
+		padding-left: 96px;
+
+		> .c-inner {
+			margin-left: -96px;
+		}
+	}
```

Comment:
Since we're going to deal with this from the JS side, would you be better off removing `treeview-nested` using `padding-left` and `margin-left`? so it is easier to deal with instead of creating a logic to add these values only at level 5 or more for example.

Reply:
@matuzalemsteles Yes, removing all the `padding-left` and `margin-left` would be the right thing to do. I think we should still keep `treeview-nested` though.

Reply:
@pat270 no problem, we can go with it.

Reply:
I found a hack I could use. We can do this without adding inline styles via JS. I'll update the pr today.

Reply:
Oh great 🤩

Reply:
@matuzalemsteles I failed, we need to go back to original plan with JS.

Reply:
😢